packages/valory/skills/abstract_round_abci/common.py

```diff
@@ -130,7 +130,7 @@ def async_act(self) -> Generator:
                 self.context.logger.info(f"Retrieved DRAND values: {observation}.")
 
         if observation:
-            payload = self.payload_class(
+            payload = self.payload_class(  # type: ignore
```

Comment:
hmm no I cannot cast it since the payload is not known at abstract base class level. There was an implicit assumption here that went unnoticed by mypy but has come to light now that we use dataclasses